committee.rst

```diff
@@ -0,0 +1,164 @@
+  The committee does not need to weigh in during this process, though the
+  expectation is that if the co-chairs abuse their power, the committee will
+  vote them out of their chairships.
+
+* **Secretary.** The committee has one secretary. The secretary is the first stop for keeping
+  our repository up-to-date and running our selection process.
+
+* **Nudger.** The committee has one nudger. The nudger's job is to make sure other
+  committee members are fulfilling their duties by nudging them (often
+  in public) to do so.
+
+The co-chairs and secretary
+positions remain with an individual until they decide to relinquish the
+role, though a majority vote among the committee can strip a member of
+their co-chair/secretary position.
+
+The nudger position rotates among members every six months.
+In the event that the nudger ends up needing nudging (by any other member
+of the committee), the nudger should step down and volunteer to do a better
+job in some future six months; in the event that they still struggle to
+stay on top of their nudging duties, it may be a sign that they are too
+busy for committee work and should step down.
+
+The secretary is in charge of managing who the nudger is, by creating
+a rotation for this position.
+
+Guidelines for Communication
+----------------------------
+
+All members of the Steering Committee agree to abide by the
+`Guidelines for Respectful Communication <https://github.com/ghc-proposals/ghc-proposals/blob/master/GRC.rst>`_. As the guidelines document
+describes, if one of us errs in our communication, please
+opt to bring up the matter with that individual directly.
+If that is not feasible, then please bring it up with the
+(co-)chair of the committee. Everyone should feel welcomed
+to contribute to the growth of GHC and Haskell, and we eagerly
+want to know when our actions cause someone not to feel that
+way.
+
+Method of Communication
+-----------------------
+
+Most official committee business takes place via the
+`ghc-steering-commitee <https://mail.haskell.org/mailman/listinfo/ghc-steering-committee>`_ mailing list. The archives of this list are public, and
+any member of the community may subscribe. It is intended that the list
+be used by the committee, though community members might be asked
+to post there during relevant conversations.
+
+Any business not on the public mailing list will be of a sensitive nature,
+most likely pertaining to individuals. In particular, discussions of selecting
+new members and of violations of the guidelines for respectful communication
+will not appear on the mailing list.
+
+Term Limits and Committee Selection
+-----------------------------------
+
+Any community thrives best by continuing to refresh itself with new members.
+Accordingly, membership on the committee comes with a three-year term.
+Toward the end of an individual's term, the secretary of
+the committee will put out a public call for nominations for people to join
```

Comment:
Sorry for re-reading this carefully only now, but I am a bit confused: Do we call for nominations when

 * any time a member's term ends
 * when the number of members drop below 9 (as written further down)
 * in both events?

If only when the number drops, that does mean that re-nomination is only possible when enough members have dropped out?

Reply:
Good point. I've amended the text to clarify that the answer is "both". It seems annoying for an existing member not to be able to stay on the committee if they and the rest of the committee wishes for them to, just because of these rules.

Reply:
If this was an IR, we’d say it has too many ad-hoc typing rules ;-)

But we can refine as we go.

Reply:
Yes, it's a bit fiddly. But I don't see a better option here, other than perhaps fixing the size of the committee, which would make this point disappear. Yet I really like the idea of giving ourselves the freedom to accept more people than we lose, when we have a pool of good willing volunteers.

Reply:
Another option might be to extend terms until enough members have their 3 year term lapsed to put the total below 9.

Or put differently: 
 * Calls for nominations happen when there are less than 9 active members who have served less than 3 years (or maybe a bit before that happens)
 * Terms end with the first call for nominations after 3 years.
 
Also, should expiring members stop being members not at 3 years exactly, but at the end of the nomination and voting period, so that the committee isn’t suddenly empty, and that rejoining members are effectively serving without interruption?

Reply:
I somehow have a hard time wrapping my head around the rules you propose. I'm inclined to keep the rules as I've written them in this regard. Do others have opinions here?

I'll clarify that departing members should stick around until their replacements arrive.

Reply:
Yeah, I have a hard time to write them down in a simple way, although I think they are, in fact, simple.

The current state of the PR has some odd behavor. For example, if we have a 11 member committee, and Joe’s term expires. It now depends soley on Joe’s whim whether there is a public nomination or not. Moreover, assume that Joe wants to stay on, then we have a round of nomination with a large committee, maybe taking on additional members. This can happen over and over again, and the committee grows out of bounds.

With my rules, the committee is bounded by “9 + number of people joining after a _single_ round of nominations”.

Let me try to rephrase the proposed rules, by distinguishing “active” members (<3 since joining or re-joining), “lame duck” members (≥ 3 years, but still active; don’t use this term in the bylaws) and “retired”. Then the rule are stated:

 * After 3 years, active members become lame ducks (but nothing happens pe se)
 * A public nomination happens when there are less than 9 active members.
 * Lame duck members retire (or rejoin) at the end of the next nomination process after they turned lamed ducks.

Is that clearer?

Reply:
Another way to think about @nomeata's rules is that we'd have fewer, larger nomination cycles as we batch up members whose terms have formally expired. I think that's probably a good thing, but haven't thought about it too carefully.

Reply:
My inclination is to favour @nomeata 's proposal.

Reply:
OK. I have updated the proposal accordingly. Please double-check to make sure it means what we described here. This is somehow trickier than I would have thought.